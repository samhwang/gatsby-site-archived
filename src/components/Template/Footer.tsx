import React from 'react';
import { OutboundLink } from 'gatsby-plugin-google-gtag';

function Footer() {
  const GitHubLink = (
    <OutboundLink href="https://github.com/samhwang">Sam Huynh</OutboundLink>
  );

  const GitHubPages = (
    <OutboundLink href="https://pages.github.com/">GitHub Pages</OutboundLink>
  );

  const GatsbyJSLink = (
    <OutboundLink href="https://www.gatsbyjs.org/">GatsbyJS.</OutboundLink>
  );
  return (
    <section className="resume-section p-3 p-lg-5 d-flex align-items-center">
      <p>
        {`Â© 2020 `}
        {GitHubLink}
        {`. Hosted on `}
        {GitHubPages}
        {`. Based on `}
        {GatsbyJSLink}
      </p>
    </section>
  );
}

export default Footer;
