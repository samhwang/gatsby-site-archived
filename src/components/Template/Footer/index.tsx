import type { FC } from 'react';
import {
  Container,
  Typography,
  makeStyles,
  createStyles,
} from '@material-ui/core';
import type { Theme } from '@material-ui/core';
import { OutboundLink } from 'gatsby-plugin-google-gtag';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    footer: {
      padding: theme.spacing(3, 2),
      marginTop: 'auto',
    },
  })
);

const Footer: FC = () => {
  const classes = useStyles();
  const GitHubLink = (
    <OutboundLink href="https://github.com/samhwang">Sam Huynh</OutboundLink>
  );

  const GitHubPages = (
    <OutboundLink href="https://pages.github.com/">GitHub Pages.</OutboundLink>
  );

  const GatsbyJSLink = (
    <OutboundLink href="https://www.gatsbyjs.org/">GatsbyJS</OutboundLink>
  );

  return (
    <footer className={classes.footer}>
      <Container maxWidth="md">
        <Typography paragraph variant="body2">
          {`Â© ${new Date().getFullYear()} `}
          {GitHubLink}
          {`. Built with `}
          {GatsbyJSLink}
          {` and hosted on `}
          {GitHubPages}
        </Typography>
      </Container>
    </footer>
  );
};

export default Footer;
