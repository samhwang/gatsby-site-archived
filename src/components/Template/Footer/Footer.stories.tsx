import Footer from '.';
import 'bootstrap/dist/css/bootstrap.css';
import '../../../assets/sass/resume.scss';

export default {
  title: 'Footer',
  component: Footer,
};

export const FooterTest = () => <Footer />;
