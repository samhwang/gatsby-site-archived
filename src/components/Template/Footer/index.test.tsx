/* eslint-disable import/no-extraneous-dependencies */
import { render } from '@testing-library/react';
import Footer from '.';

describe('Footer', () => {
  it('Should render without crashing and match snapshot', () => {
    const tree = render(<Footer />);
    expect(tree).toMatchSnapshot();
  });
});
