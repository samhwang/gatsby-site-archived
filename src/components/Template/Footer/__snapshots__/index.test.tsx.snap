// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Should render without crashing and match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="makeStyles-footer-1"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthMd"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-paragraph"
          >
            © 2021 
            <a
              href="https://github.com/samhwang"
            >
              Sam Huynh
            </a>
            . Built with 
            <a
              href="https://www.gatsbyjs.org/"
            >
              GatsbyJS
            </a>
             and hosted on 
            <a
              href="https://pages.github.com/"
            >
              GitHub Pages.
            </a>
          </p>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="makeStyles-footer-1"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthMd"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-paragraph"
        >
          © 2021 
          <a
            href="https://github.com/samhwang"
          >
            Sam Huynh
          </a>
          . Built with 
          <a
            href="https://www.gatsbyjs.org/"
          >
            GatsbyJS
          </a>
           and hosted on 
          <a
            href="https://pages.github.com/"
          >
            GitHub Pages.
          </a>
        </p>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
